spec/expression.html

```diff
@@ -119,8 +119,10 @@ <h1>Runtime Semantics: RecordPropertyDefinitionEvaluation</h1>
           1. Let _from_ be ! ToObject(_source_).
           1. Let _keys_ be ? _from_.[[OwnPropertyKeys]]().
           1. For each element _nextKey_ of _keys_ in List order, do
-            1. Let _value_ be _from_.[[Get]](_nextKey_).
-            1. Perform ? AddPropertyIntoRecordEntriesList(_entries_, _nextKey_, _value_).
+            1. Let _desc_ be ? _from_.[[GetOwnProperty]](_nextKey_).
+            1. If _desc_ is not *undefined* and _desc_.[[Enumerable]] is *true*, then
+              1. Let _value_ be ? Get(_from_, _nextKey_).

```

Comment:
This looks "correct" in terms of matching https://tc39.es/ecma262/#sec-copydataproperties but I am a bit surprised that both cases don't use desc.[[Value]]--I guess to permit own getters? Still surprising that we repeat the MOP operation.

Reply:
Yes, spread invokes getters. I think the double get (GetOwnProperty + Get) might be observable via proxies, but if we want to change it it should be first done in the object case (and then copied in this proposal).

Reply:
I agree. I think this PR is good to land.